scripts/index.ts

```diff
@@ -0,0 +1,123 @@
+import * as dotenv from "dotenv";
+import { VaultsDocument, execute, Vault } from "../.graphclient";
+import { ethers, providers, BigNumber } from "ethers";
+import { TickMath } from "@uniswap/v3-sdk";
+const { AlchemyProvider } = providers;
+import { readFileSync } from "fs";
+import { getAmountsForLiquidty } from "./liquidity-math";
+import { sqrt } from "./math-utils";
+import { liquidate } from "./bot";
+
+dotenv.config({ path: __dirname + "/.env" });
+
+const main = async () => {
+    const provider = new AlchemyProvider("goerli", process.env.ALCHEMY_KEY);
+    let cdp = new ethers.Contract(
+        process.env.CDP as string,
+        readFileSync("./scripts/abis/Vault.json", "utf-8"),
+        provider
+    );
+    const vaults: Vault[] = (await execute(VaultsDocument, {})).data.vaults;
+    const tokenPricesX96 = await getNeededPricesX96(vaults, provider, cdp);
+    const globalStabilisationFeePerUSDD = await cdp.globalStabilisationFeePerUSDD();
+    for (let i = 0; i < vaults.length; ++i) {
+        const vault = vaults[i];
+        if (await liquidationNeeded(vault, tokenPricesX96, globalStabilisationFeePerUSDD)) {
+            await liquidate(vault, cdp, provider);
+        }
+    }
```

Comment:
Let's run this in a loop, once each N seconds?

Reply:
I think the loop can be done in some deployment config. For example in cron config

Reply:
I don't think it's a good idea to run this using cron. I will create an extra overhead, reduce flexibility and make unnecessary RPC requests. So let's make this loop inside the process.